### PR TITLE
xmap and pjit integration

### PR DESCRIPTION
**Pull request by colemanliyah**

### CODE REVIEW

File: tests/compilation_cache_test.py [hunk 1]
```diff
@@ -195,6 +199,46 @@ def test_jit(self):
...
+  def test_xmap(self):
+      with tempfile.TemporaryDirectory() as tmpdir:
+          cc.initialize_cache(tmpdir)
+          def f(x):
+              return x * 2
+          devices = np.array(jax.local_devices()[:2])
+          if devices.size < 2:
+              raise SkipTest("Test requires 2 devices")
+          x = np.arange(8, dtype=np.int64).reshape((2, 2, 2))
+          with mesh(devices, ('x',)):
```
Comment (skye) [inline tests/compilation_cache_test.py:231]: I _think_ you can replace all of this with the `@jtu.with_mesh([('x', 2)])` decorator used on the pjit test. Try it?

**Review by skye — CHANGES_REQUESTED** [review]
Looks great!

**Review by skye — APPROVED** [review]
@zhangqiaorjc wanna take a look?

**Review by zhangqiaorjc — APPROVED** [review]

**Comment by gnecula** [comment]
I think that it would be nice to add a bit more descriptive titles and some description to the PRs. Otherwise somebody who sees this in a git history can easily think that the PR does something very different.

**Pull request merged**